### PR TITLE
ramips: add support for CreativeBox v1

### PR DESCRIPTION
**Pull request by hackpascal**
Hardware:
SoC: MT7621
DRAM: 512MB DDR3
Flash: 32MB SPI-NOR
WiFi 2.4GHz: MT7603 @ PCIe0
WiFi 5.8GHz: MT7612 @ PCIe1
SATA: ASM1061 @ PCIe2 (SATA + eSATA)

Interfaces:
GBE RJ45 x5
USB3.0 x1
eSATA (with USB2.0) x1
SATA x1
UART x1
I2C x1
JTAG x1

Flash instructions:
Through factory bootloader or firmware web interface

Signed-off-by: Weijie Gao <hackpascal@gmail.com>

### CODE REVIEW

File: target/linux/ramips/base-files/etc/board.d/01_leds [hunk 1]
```diff
@@ -82,6 +82,9 @@ cf-wr800n)
 	ucidef_set_led_netdev "lan" "lan" "$boardname:white:ethernet" eth0.1
 	set_wifi_led "$boardname:white:wifi"
 	;;
+xzwifi,creativebox-v1)
+	ucidef_set_led_netdev "internet" "internet" "$boardname:blue:internet" "eth0.2"
```
Comment (mkresin) [inline target/linux/ramips/base-files/etc/board.d/01_leds]: First of all, please keep the alphabetical order. Some issue in the other files.

Please use ucidef_set_led_switch instead. As soon as someone decides to use a different vlan for wan, the trigger will not work any longer.

File: target/linux/ramips/base-files/lib/upgrade/platform.sh [hunk 2]
```diff
@@ -37,6 +37,7 @@ platform_check_image() {
 	c108|\
 	carambola|\
 	cf-wr800n|\
+	xzwifi,creativebox-v1|\
```
Comment (mkresin) [inline target/linux/ramips/base-files/lib/upgrade/platform.sh]: Please rebase your PR on to of master and remove this line.

File: target/linux/ramips/dts/CreativeBox-v1.dts [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+			gpio-export,output = <1>;
+			gpios = <&gpio0 27 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
```
Comment (mkresin) [inline target/linux/ramips/dts/CreativeBox-v1.dts]: flash@

File: target/linux/ramips/dts/CreativeBox-v1.dts [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+&sdhci {
+	status = "okay";
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	mt76@0,0 {
```
Comment (mkresin) [inline target/linux/ramips/dts/CreativeBox-v1.dts]: wifi@

File: target/linux/ramips/dts/CreativeBox-v1.dts [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x0000>;
+	};
+};
+
+&pcie1 {
+	mt76@0,0 {
```
Comment (mkresin) [inline target/linux/ramips/dts/CreativeBox-v1.dts]: wifi@

File: target/linux/ramips/image/mt7621.mk [hunk 6]
```diff
@@ -121,6 +121,16 @@ define Device/mediatek_ap-mt7621a-v60
 endef
 TARGET_DEVICES += mediatek_ap-mt7621a-v60
 
+define Device/xzwifi_creativebox-v1
+  DTS := CreativeBox-v1
+  IMAGE_SIZE := $(ralink_default_fw_size_32M)
+  DEVICE_TITLE := CreativeBox v1
+  DEVICE_PACKAGES := \
+	kmod-ata-core kmod-ata-ahci kmod-mt7603 kmod-mt76x2 kmod-sdhci-mt7620 \
+	kmod-usb3
```
Comment (mkresin) [inline target/linux/ramips/image/mt7621.mk]: usb3? Not mentioned in the commit message.

File: target/linux/ramips/dts/CreativeBox-v1.dts [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0xe000>;
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "wdt", "rgmii2", "mdio";
```
Comment (mkresin) [inline target/linux/ramips/dts/CreativeBox-v1.dts]: no pin of the mdio group is used as gpio

**Review by mkresin — CHANGES_REQUESTED** [review]

File: target/linux/ramips/image/mt7621.mk (hunk 6 above)
Comment (hackpascal) [reply target/linux/ramips/image/mt7621.mk]: Hardware spec cont.

Interfaces:
GBE RJ45 x5
USB3.0 x1
eSATA(with USB2.0) x1
SATA x1
UART x1
I2C x1
JTAG x1
SPI x2 (one empty)

File: target/linux/ramips/dts/CreativeBox-v1.dts (hunk 7 above)
Comment (hackpascal) [reply target/linux/ramips/dts/CreativeBox-v1.dts]: a hidden mistake.
The gsw_mt7621 sets the pinmux of mdio explictly which makes this pinctrl useless.

**Comment by hackpascal** [comment]
@mkresin done for everything

**Comment by rmilecki** [comment]
This seems to be adding
0001-ramips-add-support-for-CreativeBox-v1.patch
in the project's root directory.

I guess that patch file is not needed at all since you also add
target/linux/ramips/dts/CreativeBox-v1.dts

File: target/linux/ramips/dts/CreativeBox-v1.dts [hunk 8]
```diff
@@ -0,0 +1,172 @@
+/dts-v1/;
```
Comment (rmilecki) [inline target/linux/ramips/dts/CreativeBox-v1.dts]: Please include licensing info for your DTS file, see
https://openwrt.org/submitting-patches#dts_checklist
e.g. a line like:
`// SPDX-License-Identifier: GPL-2.0-or-later OR MIT`
above the `/dts-v1/;`

Comment (hackpascal) [reply target/linux/ramips/dts/CreativeBox-v1.dts]: forgot that.

**Comment by mkresin** [comment]
merged, thanks!

**Comment by adrianschmutzler** [comment]
Hi, I'm currently reviewing the MAC address assignment on ramips target.
For your device, you are currently using MAC address from <&factory 0xe000> and then just calculate WAN address by adding +1 to that.
However, there might be another MAC address stored on flash for your device in <&factory 0xe006>.

Can you check that? If yes, I would also be interested in which of the two addresses should be "lan" and which "wan" then ...

**Pull request closed without merging**